smsdk/smsdk_entities/machine_type/machinetypeV1.py

```diff
@@ -0,0 +1,137 @@
+from typing import List
+import json
+from datetime import datetime, timedelta
+
+try:
+    import importlib.resources as pkg_resources
+except ImportError:
+    # Try backported to PY<37 `importlib_resources`.
+    import importlib_resources as pkg_resources
+
+
+import numpy as np
+
+from smsdk.tool_register import SmsdkEntities, smsdkentities
+from smsdk.utils import module_utility, check_kw
+from smsdk import config
+from smsdk.ma_session import MaSession
+
+ENDPOINTS = json.loads(pkg_resources.read_text(config, "api_endpoints.json"))
+
+@smsdkentities.register("machine_type_v1")
+class MachineType(SmsdkEntities, MaSession):
```

Comment:
Lot of duplicate code in all these  files which can be moved to a Parent class and reuse from there